tests/vm_migration_test.go

```diff
@@ -0,0 +1,128 @@
+package tests_test
+
+import (
+	"flag"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/errors"
+	kubev1 "k8s.io/client-go/pkg/api/v1"
+	metav1 "k8s.io/client-go/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/util/json"
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"net/http"
+	"time"
+)
+
+var _ = Describe("VmMigration", func() {
+
+	flag.Parse()
+
+	restClient, err := kubecli.GetRESTClient()
+	tests.PanicOnError(err)
+	var vm *v1.VM
+	var migration *v1.Migration
+
+	BeforeEach(func() {
+		vm = tests.NewRandomVM()
+		migration = tests.NewMigrationForVm(vm)
+
+		tests.MustCleanup()
+	})
+
+	Context("New Migration given", func() {
+
+		It("Should be No migrations", func() {
+
+			b, err := restClient.Get().Resource("migrations").Namespace(api.NamespaceDefault).Name(migration.ObjectMeta.Name).DoRaw()
+			Expect(err).ToNot(BeNil())
+			status := metav1.Status{}
+			err = json.Unmarshal(b, &status)
+			Expect(err).To(BeNil())
+			Expect(status.Code).To(Equal(int32(http.StatusNotFound)))
+
+			r, err := restClient.Get().Resource("migrations").Namespace(api.NamespaceDefault).Name(migration.ObjectMeta.Name).Do().Get()
+			var se *errors.StatusError = err.(*errors.StatusError)
+			Expect(se.ErrStatus.Code).To(Equal(int32(404)))
+			Expect(r).To(BeNil())
+
+		})
+
+		It("Should fail if the VM does not exist", func() {
+			r, err := restClient.Get().Resource("vms").Namespace(api.NamespaceDefault).Name(vm.ObjectMeta.Name).Do().Get()
+			var se *errors.StatusError = err.(*errors.StatusError)
+			Expect(se.ErrStatus.Code).To(Equal(int32(404)))
```

Comment:
We have a matching library for kubernetes response objects in `pkg/virt-api/kubeproxy/rest/matcher_test.go`.

It allows checks like:

```golang
result := restClient.Get().Resource("vms").Namespace(api.NamespaceDefault).Name(vm.ObjectMeta.Name).Do()
 
Expect(result).To(HaveStatusCode(http.StatusOK))
Expect(result).To(HaveBodyEqualTo(expectedVM))
```

Maybe you can move the matcher to a shared location and start using it here too?